andela/ah-backend-athena
# #162163273 Users should be able to favorite articles.

**What does this PR do?**
This Pull Request enables a user to favorite an article

**Description of tasks**
Below are the endpoints that enable the user have the access to favorite or unfavorite an article
`POST` ` /api/articles/<slug>/favorite/`
`DELETE` ` /api/articles/<slug>/favorite/`

**How to manually test it**

- clone the repository

- install the dependencies in the `requirements.txt` file

- the `POST` ` /api/articles/<slug>/favorite/` should return
```
{
  "article": {
    "slug": "how-to-train-your-dragon",
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian",
    "tagList": ["dragons", "training"],
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "favorited": true,
    "favoritesCount": 2,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    }
  }
}
```

- the `DELETE` ` /api/articles/<slug>/favorite/` should return
```
{
 "message":{
  "body":[
    "article has been unfavorited"
  ]
 }
}
```

**Relevant pivotal tracker stories**
#162163273
